### PR TITLE
Windows builder docker

### PR DESCRIPTION
**Pull request by cakekoa**
- Pull in the pywine docker image build files
- Update the builder to properly use the shared wine prefix as a non-root user

### CODE REVIEW

**Comment by mssalvatore** [comment]
I'm assuming all of the files in `pywine/`, and perhaps some others, are copied from somewhere else? Could we include that repo as a git submodule instead of copying the files? It would make us better able to keep synced with upstream.

**Comment by cakekoa** [comment]
@mssalvatore  Yes, both `pywine` and `wine-docker` are from repos hosted on github. I think the only difference is I've modified the `FROM` line in pywine to expect a different tag. I should be able to include `wine-docker` as a submodule. I could probably include `pywine` as a submodule by either forking the original and adding my change, or just adding my change in the build script.

**Comment by cakekoa** [comment]
I've added those repos as submodules and updated the build script. I think I should squash these commits down before merging

File: docker/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+SCRIPT_DIR="$(realpath $(dirname $BASH_SOURCE[0]))"
+
+cd "$SCRIPT_DIR"/wine-docker
+docker build --platform linux/amd64 -t wine-docker:devel --build-arg WINE_FLAVOUR=devel .
```
Comment (mssalvatore) [inline docker/build.sh]: Consider naming this image something more unique to avoid the potential for collisions.

File: docker/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+cd "$SCRIPT_DIR"/pywine
+sed 's|FROM tobix/wine:devel|FROM wine-docker:devel|' Dockerfile > Dockerfile.modified
```
Comment (mssalvatore) [inline docker/build.sh]: We need some kind of sanity check to ensure that the dockerfile was successfully modified. For example, if the original file is changed to use `tobix2/wine:devel`, this `sed` command will have no effect and our build won't be based on the correct image.

A simple solution might be to compare hashes of the two files. Another solution might be to use `diff -q Dockerfile Dockerfile.modified` and check that it's return code (`$?`) is non-zero.

File: docker/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+docker build --platform linux/amd64 -t pywine:3.11 --build-arg PYTHON_VERSION=3.11.9 -f Dockerfile.modified .
```
Comment (mssalvatore) [inline docker/build.sh]: Again, a more unique name/tag might be advisable.

File: docker/README.md
Comment (mssalvatore) [inline docker/README.md:1]: Can we also document how to upload this to dockerhub?

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**